ontotext-yasgui-web-component/src/components/ontotext-yasgui-web-component/ontotext-yasgui-web-component.tsx

```diff
@@ -197,12 +197,23 @@ export class OntotextYasguiWebComponent {
    * Allows the client to init the editor using a query model. When the query and query name are
    * found in any existing opened tab, then it'd be focused. Otherwise a new tab will be created and
    * initialized using the provided query model.
-   * @param query The query model.
+   * @param queryModel The query model.
    */
   @Method()
-  openTab(query: TabQueryModel): Promise<void> {
-    this.ontotextYasgui.openTab(query);
-    return Promise.resolve();
+  openTab(queryModel: TabQueryModel): Promise<void> {
+    // While this does the job in this particular method, we definitely need a more general approach
+    // for handling the fact that there is a chance for the client to hit the problem where when the
+    // OntotextYasgui instance is created and returned the wrapped Yasgui instance might not be yet
+    // initialized.
+    return new Promise((resolve) => {
+      const timer = setInterval(() => {
+        if (this.ontotextYasgui.getInstance()) {
+          this.ontotextYasgui.openTab(queryModel);
+          clearInterval(timer);
```

Comment:
It might be good if we clear the interval if the instance is not found for a certain amount of time. For example 1 minute.